metricproviders/graphite/api.go

```diff
@@ -73,7 +73,7 @@ func (api APIClient) Query(quer string) ([]dataPoint, error) {
 
 func (api APIClient) trimQuery(q string) string {
 	space := regexp.MustCompile(`\s+`)
-	return space.ReplaceAllString(q, " ")
+	return space.ReplaceAllString(q, "")
```

Comment:
One can possibly argue that this function name doesn't reflect its purpose: it doesn't **just** trim the whitespace, but also can modify the input string content in the middle.
Also, compiling regular expression every time this function is called is sub-optimal.
Considering the above, it can be rewritten to something like this:

```golang
var space = regexp.MustCompile(`\s+`)

func (api APIClient) sanitizeQuery(q string) string {
	return space.ReplaceAllLiteralString(q, "")
}
```

Reply:
Yea I like this change.

Reply:
> Yea I like this change.

Thank you, I've uploaded this as a separate commit.